roc.go

```diff
@@ -0,0 +1,154 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+// SortWeightedBool rearranges the data in y along with their
```

Comment:
This also needs to take in weights to be SortWeightedBool :) 

Couple of bikesheds. 1) Could you make it `x`instead of `y` to match SortWeighted. 2) Could you move it to stat.go?